src/aio.rs

```diff
@@ -430,7 +430,7 @@ fn get_socket_addrs(host: &str, port: u16) -> RedisResult<SocketAddr> {
 pub trait ConnectionLike {
     /// Sends an already encoded (packed) command into the TCP socket and
     /// reads the single response from it.
-    fn req_packed_command<'a>(&'a mut self, cmd: &'a Cmd) -> RedisFuture<'a, Value>;
+    fn req_packed_command<'a>(&'a mut self, cmd: &'a [u8]) -> RedisFuture<'a, Value>;
```

Comment:
Instead of changing this signature you should be able to update the `cursor` inside the `Cmd`. Remove the cursor inside `AsyncIter` and update the one in `Cmd` instead.

(There are downstream crates like https://github.com/marwes/redis-cluster-rs which rely on `Cmd` being passed in.